client/icons.go

```diff
@@ -0,0 +1,65 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2015 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package client
+
+import (
+	"fmt"
+	"io/ioutil"
+	"mime"
+	"net/http"
+)
+
+// Icon represents the icon of an installed Package
+type Icon struct {
+	Extension string
+	Content   []byte
+}
+
+// Icon returns the Icon belonging to an installed Package
+func (c *Client) Icon(pkgID string) (Icon, error) {
+	const errPrefix = "cannot retrieve icon"
+
+	response, err := c.raw("GET", fmt.Sprintf("/1.0/icons/%s/icon", pkgID), nil)
+	if err != nil {
+		return Icon{}, fmt.Errorf("%s: failed to communicate with server: %s", errPrefix, err)
+	}
+	defer response.Body.Close()
+
+	if response.StatusCode != http.StatusOK {
+		return Icon{}, fmt.Errorf("%s: Not Found", errPrefix)
+	}
+
+	extensions, err := mime.ExtensionsByType(response.Header.Get("Content-Type"))
```

Comment:
Cool. Sending `Content-Disposition: filename=myicon.png` rather than inferring the file extension from the MIME-type would be a lot cleaner.

I can take this on if you've already got your plate full.